pkg/network/usm/utils/connection_rollup.go

```diff
@@ -210,34 +188,54 @@ func (c *ConnectionAggregator) ClearEphemeralPort(key types.ConnectionKey) types
 		return key
 	}
 
-	normalizedKey, _ := normalizeByIP(key)
-	ips, ports := split(normalizedKey)
-	savedPorts, ok := c.data[ips]
-	if !ok || savedPorts.ephemeralSide == ephemeralPortUnknown ||
-		(ports.a != savedPorts.a && ports.b != savedPorts.b) {
+	ips, ports, flipped := splitKey(key)
+	savedPorts, ok := c.findSimilar(ips, ports)
+	if !ok || savedPorts.serverSide == serverPortUnknown {
 		// We either haven't seen at this connection, or were not able to
 		// determine the ephemeral port side. In this case we return the key
 		// completely unmodified.
 		return key
```

Comment:
Shouldn't we indicate that something is wrong in this scenario? The caller won't be aware that we've returned the unmodified key, which contradicts the function's supposed action of clearing the ephemeral port

Reply:
Yeah, at the very least I think I should mention in the method documentation that the function may or may not modify the key.
I wouldn't say that there is anything _wrong_ in returning a key that is not modified. For example,
If we our event stream sees only one connection like:
```
1.1.1.1:port-x,2.2.2.2:port-y
```
we can't really say which side is which (unless we rely on other heuristics like connection direction, system configuration etc, but those have proven to be unreliable over time, so I'm avoiding them here)  so we wouldn't clear any port numbers, which is fine.
I guess a compromise would be perhaps returning a `modified bool` as opposed to an `error` to let the caller know what's the case?

Reply:
Done